Sources/Utility/Process.swift

```diff
@@ -0,0 +1,295 @@
+        sigdelset(&mostSignals, SIGSTOP)
+        posix_spawnattr_setsigdefault(&attributes, &mostSignals)
+      #else
+        // On Linux, this can only be used to reset signals that are legal to
+        // modify, so we have to take care about the set we use.
+        var mostSignals = sigset_t()
+        sigemptyset(&mostSignals);
+        for i in 1 ..< SIGUNUSED {
+            if i == SIGKILL || i == SIGSTOP {
+                continue
+            }
+            sigaddset(&mostSignals, i)
+        }
+        posix_spawnattr_setsigdefault(&attributes, &mostSignals)
+      #endif
+
+        // Establish a separate process group.
+        posix_spawnattr_setpgroup(&attributes, 0)
+
+        // Set the attribute flags.
+        var flags = POSIX_SPAWN_SETSIGMASK | POSIX_SPAWN_SETSIGDEF
+        flags |= POSIX_SPAWN_SETPGROUP
+
+        // Close all other files by default.
+        //
+        // FIXME: Note that this is an Apple-specific extension, and we will have to
+        // do something else on other platforms (and unfortunately, there isn't
+        // really an easy answer other than using a stub executable).
+      #if os(macOS)
+        flags |= POSIX_SPAWN_CLOEXEC_DEFAULT
+      #endif
+
+        posix_spawnattr_setflags(&attributes, Int16(flags))
+
+        // Setup the file actions.
+      #if os(macOS)
+        var fileActions: posix_spawn_file_actions_t? = nil
+      #else
+        var fileActions = posix_spawn_file_actions_t()
+      #endif
+        posix_spawn_file_actions_init(&fileActions)
+
+        // Workaround for https://sourceware.org/git/gitweb.cgi?p=glibc.git;h=89e435f3559c53084498e9baad22172b64429362
+        let devNull = strdup("/dev/null")
+        defer { free(devNull) }
+        // Open /dev/null as stdin.
+        posix_spawn_file_actions_addopen(&fileActions, 0, devNull, O_RDONLY, 0)
+
+        var rv = libc.pipe(&saveOutputPipe)
+        guard rv == 0 else {
+            throw SystemError.pipe(rv)
+        }
+
+        if outputStream != nil {
+            // Open the write end of the pipe as stdout and stderr, if desired.
+            posix_spawn_file_actions_adddup2(&fileActions, saveOutputPipe[1], 1)
+            posix_spawn_file_actions_adddup2(&fileActions, saveOutputPipe[1], 2)
+
+            // Close the other ends of the pipe.
+            posix_spawn_file_actions_addclose(&fileActions, saveOutputPipe[0])
+            posix_spawn_file_actions_addclose(&fileActions, saveOutputPipe[1])
+        } else {
+            posix_spawn_file_actions_adddup2(&fileActions, 1, 1)
+            posix_spawn_file_actions_adddup2(&fileActions, 2, 2)
+        }
+
+        let argv: [UnsafeMutablePointer<CChar>?] = arguments.map{ $0.withCString(strdup) }
+        defer { for case let arg? in argv { free(arg) } }
+        
+        let env: [UnsafeMutablePointer<CChar>?] = environment.map{ "\($0.0)=\($0.1)".withCString(strdup) }
+        defer { for case let arg? in env { free(arg) } }
```

Comment:
`var argv` + argv.append(nil) would be mildly more efficient than our `argv + [nil]` (used later)